src/main/java/appeng/parts/p2p/PartP2PFluids.java

```diff
@@ -183,33 +232,43 @@ public int fill( final EnumFacing from, final FluidStack resource, final boolean
 		return used;
 	}
 
-	private Stack<PartP2PLiquids> getDepth()
+	@Override
+	public FluidStack drain( FluidStack resource, boolean doDrain )
 	{
-		Stack<PartP2PLiquids> s = DEPTH.get();
+		return null;
+	}
+
+	@Override
+	public FluidStack drain( int maxDrain, boolean doDrain )
+	{
+		return null;
+	}
+
+	private Stack<PartP2PFluids> getDepth()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "Stack" by an unsynchronized one such as "Deque". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1149)